sphinxlint/checkers.py

```diff
@@ -118,8 +118,12 @@ def check_default_role(file, lines, options=None):
             before_match = line[: match.start()]
             after_match = line[match.end() :]
             stripped_line = line.strip()
-            if (stripped_line.startswith("|") and stripped_line.endswith("|") and
-                stripped_line.count("|") >= 4 and "|" in match.group(0)):
+            if (
+                stripped_line.startswith("|")
+                and stripped_line.endswith("|")
+                and stripped_line.count("|") >= 4
+                and "|" in match.group(0)
+            ):
```

Comment:
This change seems unnecessary, since it takes 3x the number of lines and makes the condition overall less readable.

Reply:
Interesting, I find the extra lines more readable.

It gives us one condition per line, making it easier to compare each one just by reading up/down; rather than a 2x2 grid of two conditions per line, needing to read left/right/up/down to compare.

Also, the old one is less readable in a diff, as it doesn't fit and wraps:

<img width="1095" alt="image" src="https://github.com/sphinx-contrib/sphinx-lint/assets/1324225/fa4c62b9-6d86-42a8-a97d-d04a22006948">

Reply:
I think this was done by `black`, so there's probably nothing we can do.

<details>
<summary>However, if you want to know why I find it change worse...</summary>

I intentionally wrote it that way for several reasons:
1. This check is fairly inconsequential, and doesn't need to be prominent.  Before it took 3/18 lines, whereas now it takes 6/22 lines, meaning that now several lines are used by something not particularly important -- right in the middle of the function.
2. Two lines are wasted by `if (` and `):`: this could have been avoided while remaining PEP-8 compliant.
3. Wasting lines like this means longer files, less code visible in a single screen, and more scrolling, making the file more difficult to navigate/edit.
4. In that side-by-side diff it looks indeed less readable, but it also appears to have two ~60-char wide columns whereas the lines themselves are 82 char -- within black's limit and a bit over the PEP-8 limit, and still shorter than other lines in the file.
5. In addition to this, each line of the condition had a different reason to be grouped: the first checked that the line was within '|'s (one at the start/left, one at the end/right), whereas the second line checked the number of `|`.
6. In principle I agree that having one condition per line is better (even though often it's not worth the extra lines), however the placement of the `and`s makes things worse since the first 2 conditions are not aligned, to the point that I initially thought the first condition was repeated twice by mistake and was going to report it.  FTR this was changed in the PEP-8 somewhat recently, for reasons that I don't agree with, but I didn't bring it up (yet?).

I also understand that all these points are somewhat minor and that by looking at the example in a void it might seem better, but the combination of them in the bigger context makes the change worse IMHO.  It might also just be me and the way I read/scan code and prioritize things.

If `black` changes were a bit more conservative and a bit more economic on vertical space, I would be fine with it, but unfortunately now it's probably too late to change that, even if I managed to convince the authors.  In addition, an automated tool, while it's probably good for most cases, can't really understand or handle certain subtleties.  That's why the PEP-8 itself says:

> However, know when to be inconsistent – sometimes style guide recommendations just aren’t applicable. When in doubt, use your best judgment. 

and 

> Some other good reasons to ignore a particular guideline:  When applying the guideline would make the code less readable, even for someone who is used to reading code that follows this PEP.

The good news is that I already wanted to refactor that check, put it in a function, and use it in the other functions that still have a more naive check.  Doing that will solve the first 3 points above.

</details>

Reply:
> put it in a function

Nice! That's what I had in mind while reading the above arguments.

So I just did it. The next step is obviously to handle tables, but not in this PR.